app/src/main/java/com/team41/boromi/dbs/BookDB.java

```diff
@@ -0,0 +1,116 @@
+package com.team41.boromi.dbs;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.google.gson.Gson;
+import com.team41.boromi.models.Book;
+import com.team41.boromi.models.User;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import static com.team41.boromi.constants.CommonConstants.DB_TIMEOUT;
+
+@Singleton
+public class BookDB {
+
+	private final static String TAG = "BookDB";
+	private final String DB_COLLECTION = "books";
+	private final FirebaseFirestore db;
+	private final CollectionReference booksRef;
+
+	private final Gson gson = new Gson();
+
+	@Inject
+	public BookDB(FirebaseFirestore db) {
+		this.db = db;
+		booksRef = db.collection(DB_COLLECTION);
+	}
+
+	/**
+	 * Attempts to get all books owned by the user
+	 * @param username The username of the authenticated username
+	 * @return A list of all the books owned by the user
+	 */
+	public ArrayList<Book> getUsersOwnedBooks(String username) {

```

Comment:
By looking at the UML, Isn't this part of the book controller?

Reply:
Yeah there's been a slight change to that architecture, you can refer to general in the discord. It's okay if it doesn't follow the UML. imo the uml shouldn't even have all those methods, I just put them in to satisfy the prof haha. Because software changes a lot during development you can't reasonably plan everything